src/embed/init.ts

```diff
@@ -53,49 +53,62 @@ window['_dlo_rulesFromOpsTeam'] = [
-
-    const prop = (window as { [key: string]: any })[propName];
-    if (Array.isArray(prop) === false) {
-      Logger.getInstance().warn(LogMessageType.RuleInvalid,
-        { property: prop, reason: 'Rules list must be an array' });
-      return;
-    }
-
-    prop.forEach((rule: any) => {
-      results.push(rule);
+  try {
+    const startTime = Date.now();
+    const results: any[] = [];
+    Object.getOwnPropertyNames(window).forEach((propName) => {
+      if (propName.startsWith('_dlo_rules') === false) return;
+
+      const prop = (window as { [key: string]: any })[propName];
+      if (Array.isArray(prop) === false) {
+        Logger.getInstance().warn(LogMessageType.RuleInvalid,
+          { property: prop, reason: 'Rules list must be an array' });
+        return;
+      }
+
+      prop.forEach((rule: any) => {
+        results.push(rule);
+      });
     });
-  });
-  return results;
+    Logger.getInstance().info(`DLO rules processing time (ms): ${startTime - Date.now()}`);
+    return results;
+  } catch (err) {
+    Logger.getInstance().error(`Error collecting DLO rules: ${err}`);
+    return [];
+  }
 }
 
 function _dlo_initializeFromWindow() {
-  const win = (window as { [key: string]: any });
+  try {
+    const startTime = Date.now();
+    const win = (window as { [key: string]: any });
 
-  if (win._dlo_observer) {
-    Logger.getInstance().warn(LogMessageType.ObserverMultipleLoad);
-    return;
-  }
+    if (win._dlo_observer) {
+      Logger.getInstance().warn(LogMessageType.ObserverMultipleLoad);
+      return;
+    }
 
-  // Read rules
-  const rules = _dlo_collectRules();
-  if (rules.length === 0) {
-    Logger.getInstance().warn(LogMessageType.ObserverRulesNone);
-  }
+    // Read rules
+    const rules = _dlo_collectRules();
+    if (rules.length === 0) {
+      Logger.getInstance().warn(LogMessageType.ObserverRulesNone);
+    }
 
-  win._dlo_observer = new DataLayerObserver({
-    appender: win._dlo_appender || undefined,
-    beforeDestination: win._dlo_beforeDestination || undefined,
-    logLevel: win._dlo_logLevel || undefined,
-    previewMode: win._dlo_previewMode === true,
-    previewDestination: win._dlo_previewDestination || undefined,
-    readOnLoad: win._dlo_readOnLoad === true,
-    validateRules: win._dlo_validateRules === true,
-    urlValidator: win._dlo_urlValidator || undefined,
-    rules,
-  });
+    win._dlo_observer = new DataLayerObserver({
+      appender: win._dlo_appender || undefined,
+      beforeDestination: win._dlo_beforeDestination || undefined,
+      logLevel: win._dlo_logLevel || undefined,
+      previewMode: win._dlo_previewMode === true,
+      previewDestination: win._dlo_previewDestination || undefined,
+      readOnLoad: win._dlo_readOnLoad === true,
+      validateRules: win._dlo_validateRules === true,
+      urlValidator: win._dlo_urlValidator || undefined,
+      rules,
+    });
+    Logger.getInstance().info(`DLO init (ms): ${startTime - Date.now()}`);
```

Comment:
Previous comments apply here: debug, timing context, externalize the string, and pass err as reason.message.